src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,167 @@
+package firrtl.passes
+
+import scala.collection.mutable.{ArrayBuffer,HashMap}
+import AnalysisUtils._
+import firrtl.WrappedExpression._
+import firrtl.ir._
+import firrtl._
+import firrtl.Utils._
+import firrtl.Mappers._
+
+case class AppendableInfo(fields: Map[String,Any]) extends Info {
+  def append(a: Map[String,Any]) = this.copy(fields = fields ++ a)
+  def append(a: (String,Any)): AppendableInfo = append(Map(a))
+  def get(f: String) = fields.get(f)
+  override def equals(b: Any) = b match {
+    case i: AppendableInfo => fields - "info" == i.fields - "info"
+    case _ => false
+  }
+}
+
+object AnalysisUtils {
+
+  def getConnects(m: Module) = {
+    val connects = HashMap[String, Expression]()
+    def getConnects(s: Statement): Unit = s match {
+      case s: Connect =>
+        connects(s.loc.serialize) = s.expr
+      case s: DefNode =>
+        connects(s.name) = s.value
+      case s: Block =>
+        s.stmts foreach getConnects
+      case _ =>
+    }
```

Comment:
What do you mean by conditionals? = Whens or Muxes? 
I forget, but I was hoping at least for my pass, that by the mid to low firrtl transform, When's should've already been converted to Mux's? This makes this utility function not super meaningful at high firrtl, but I wasn't sure how to deal with that (and this particular function was pulled from InferReadWrite). 

Also, can you explain what "ValidIf" is? I'm not clear about it from what code I've read.